GitCommands/Git/GitModule.cs

```diff
@@ -2725,10 +2725,20 @@ public bool IsDirtyDir()
                 : null;
         }
 
+        /// <summary>
+        /// Get the file contents for the HEAD commit
+        /// </summary>
+        /// <param name="file">The Git status item</param>
+        /// <returns>An awaitable task with the requested file contents.</returns>
         public async Task<string?> GetFileContentsAsync(GitItemStatus file)
         {
-            var contents = new StringBuilder();
+            if (!file.IsTracked || (file.Staged == StagedStatus.Index && file.IsNew))
+            {
+                // File is not in Git
+                return null;
```

Comment:
return Task.FromResult(null)

Reply:
The task is not null, it returns a null string. What is the point of writing that with more text?

Reply:
If task is not null it is ok.
I'll take a closer look. 😄

Reply:
I don't know why we are using `ConfigureAwaitRunInline()` there. But we could try to get the content in parallel for two files.

Reply:
This is in #9058, I based this PR on #9058 to prepare the merge of this, so discussed there.

Reply:
My first reaction was too to suggest `Task.FromResult(null)`. But since there's an `await` below, this is not necessary, and distorts the intent.